repair/row_level.cc

```diff
@@ -491,7 +491,10 @@ using is_local_reader = bool_class<class is_local_reader_tag>;
     future<mutation_fragment_opt>
     read_mutation_fragment() {
         ++_reads_issued;
-        return _reader(db::no_timeout).then([this] (mutation_fragment_opt mfopt) {
+        // Use a very long timeout for the reader to break out any eventual
+        // deadlock within the reader. Thirty minutes should be more than
+        // enough to read a single mutation fragment.
+        return _reader(db::timeout_clock::now() + std::chrono::minutes(30)).then([this] (mutation_fragment_opt mfopt) {
             ++_reads_finished;
```

Comment:
This is not a fix. It's a band aid. What can we do to really fix it?

If the deadlock reproduces consistently, repair will never finish.

If we apply the band aid, at a minimum we should report it loudly, not as yet another timeout, so we know it happened. And maybe dump diagnostics.

Reply:
If we see the timeout exception in repair, we know it is from here because this is the only place we can timeout.